File: fastify_server/src/addons/environment.addon.ts
import { pathToFileURL } from "node:url";
import type { Addon } from "../registry/registry.js";
import { createLoaderReport } from "../contract/index.js";
import { createLoaderLogger } from "../registry/loader_logger.js";
import { discoverFiles } from "./_discover.js";

const ENV_SUFFIXES = [".env.mjs", ".env.js"] as const;

export const environmentAddon: Addon = {
  name: "environment",
  priority: 10,
  async run(_server, config, ctx) {
    const report = createLoaderReport("environment");
    const log = createLoaderLogger("environment", ctx.logger, report);

    for (const dir of config.paths.environment) {
      let result;
      try {
        result = await discoverFiles(dir, ENV_SUFFIXES);
      } catch (err) {
        log.failed("discover", dir, err);
        continue;
      }
      log.scanDir(dir, result.matched.length, result.ignored.length);
      for (const p of result.ignored) log.ignored(p);
      report.discovered += result.matched.length;

      for (const file of result.matched) {
        try {
          await import(pathToFileURL(file).href);
          report.imported += 1;
          report.registered += 1;
          log.loaded(file);
        } catch (err) {
          log.failed("import", file, err);
        }
      }
    }
    return report;
  },
};

export default environmentAddon;

File: fastapi_server/fastapi_server/bootstrap/addons/environment_addon.py
from __future__ import annotations

from fastapi_server.bootstrap.contract import create_loader_report
from fastapi_server.bootstrap.registry.registry import Addon
from fastapi_server.bootstrap.registry.loader_logger import create_loader_logger
from fastapi_server.bootstrap.addons._discover import discover_files, import_file

_SUFFIXES = (".env.py",)


def _run(_server, config, ctx):
    report = create_loader_report("environment")
    log = create_loader_logger("environment", ctx.logger, report)

    for d in config.paths.environment:
        try:
            result = discover_files(d, _SUFFIXES)
        except Exception as err:  # noqa: BLE001
            log.failed("discover", d, err)
            continue
        log.scan_dir(d, len(result.matched), len(result.ignored))
        for p in result.ignored:
            log.ignored(p)
        report.discovered += len(result.matched)

        for file in result.matched:
            try:
                import_file(file, module_name_prefix="polyglot_env")
                report.imported += 1
                report.registered += 1
                log.loaded(file)
            except Exception as err:  # noqa: BLE001
                log.failed("import", file, err)
    return report


environment_addon = Addon(name="environment", priority=10, run=_run)
